qqqwww123cdv/hometask-3/rainfall.rb

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+def data_get
+  file = File.new('data.txt')
+  file.read
+end
+
+def weather_data(town, data)
+  data.split("\n").each do |i|
+    if i.include? town
+      str = i.scan(/\d*\.\d/).map(&:to_f)
+      return str
+    end
+  end
+end
+
+def mean(town, data)
+  return -1 unless data.match "#{town}:"
+
+  weather_data(town, data).sum / 12
+end
+
+def variance(town, data)
+  return -1 unless data.match "#{town}:"
+
+  weather_data(town, data).map { |i| (mean(town, data) - i)**2 }.sum / 12
```

Comment:
a variable's name `i` as an array element does not feel good, how about something like `data_row`?

Reply:
also, let's take in account that you call `mean` on each iteration, let's do something with this 🙂